### PR TITLE
Normalize share and volume creation

### PR DESCRIPTION
**Pull request by xhernandez**
This patch normalizes the way shares and volumes are created. Till now, the volumes were created during installation of the backend, while shares were created during Samba installation. This caused some difficulties.

Now the volume is created just before creating the share which seems more logical. Some differences between backends have also been removed so that it's more uniform.

This need appeared as part of addressing the comments for #24.

### CODE REVIEW

File: playbooks/ansible/roles/samba.setup/tasks/main.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 - name: Create Samba share directories
   file:
-    path: "{{ config.paths.mount }}/{{ item.cluster_volume }}"
+    path: "{{ config.paths.mount }}/{{ item }}"
```
Comment (anoopcs9) [inline playbooks/ansible/roles/samba.setup/tasks/main.yml:9]: Here `{{ item }}` is a dictionary, right?

Comment (xhernandez) [reply playbooks/ansible/roles/samba.setup/tasks/main.yml:9]: Yes

Comment (xhernandez) [reply playbooks/ansible/roles/samba.setup/tasks/main.yml:9]: True. I'll fix it.

File: playbooks/ansible/roles/samba.setup/tasks/main.yml [hunk 2]
```diff
@@ -47,28 +47,36 @@
...
-        volume: "{{ item.cluster_volume }}"
-        path: "{{ config.paths.mount }}/{{ item.cluster_volume }}"
-        file: "{{ config.paths.samba.etc }}/smb.shares/{{ name }}.conf"
-        options: "{{ item.samba.options }}"
-      with_items: "{{ samba_shares }}"
+- name: Create mount directories
+  file:
+    path: "{{ config.paths.mount }}/{{ item.name }}"
+    state: directory
+  with_items: "{{ samba_shares }}"
```
[the remark this answers is not in the document]
Comment (xhernandez) [reply playbooks/ansible/roles/samba.setup/tasks/main.yml:54]: Yes. I overlooked that one.

**Review by anoopcs9 — APPROVED** [review]
lgtm.

**Review by Shwetha-Acharya — APPROVED** [review]

**Pull request merged**